### PR TITLE
[GHA][PT FE] Enable topk layer tests

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Enable topk tests*

### Tickets:
 - *CVS-115085*

### CODE REVIEW

**Review by eaidova — APPROVED** [review]

**Comment by mvafin** [comment]
Layer tests for Win didn't run for some reason.

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -593,9 +593,7 @@ jobs:
           # layer test requirements
           python3 -m pip install -r ${{ env.LAYER_TESTS_INSTALL_DIR }}/requirements.txt
 
-      # Ticket - 115085
       - name: PyTorch Layer Tests
-        if: ${{ 'false' }}
         shell: cmd
         run: |
           python3 -m pytest ${{ env.LAYER_TESTS_INSTALL_DIR }}/pytorch_tests -n logical -m precommit --junitxml=${{ env.INSTALL_TEST_DIR }}/TEST-pytorch.xml
```
Comment (mvafin) [inline .github/workflows/windows.yml:599]:
```suggestion
          python3 -m pytest ${{ env.LAYER_TESTS_INSTALL_DIR }}/pytorch_tests -v -m precommit --junitxml=${{ env.INSTALL_TEST_DIR }}/TEST-pytorch.xml
```

File: .github/workflows/windows.yml [hunk 2]
```diff
@@ -593,12 +593,10 @@ jobs:
           # layer test requirements
           python3 -m pip install -r ${{ env.LAYER_TESTS_INSTALL_DIR }}/requirements.txt
 
-      # Ticket - 115085
       - name: PyTorch Layer Tests
```
Comment (mvafin) [inline .github/workflows/windows.yml:596]:
```suggestion
      - name: PyTorch Layer Tests
        if: fromJSON(needs.smart_ci.outputs.affected_components).ONNX_FE.test
```

File: .github/workflows/windows.yml [hunk 3]
```diff
@@ -593,12 +593,11 @@ jobs:
...
-        if: ${{ 'false' }}
+        if: fromJSON(needs.smart_ci.outputs.affected_components).PT_FE.test
```
Comment (mvafin) [inline .github/workflows/windows.yml:597]:
```suggestion
```

File: src/frontends/pytorch/src/op/set_item.cpp [hunk 4]
```diff
@@ -22,11 +22,13 @@ OutputVector translate_set_item(const NodeContext& context) {
     auto zero = context.mark_node(v0::Constant::create(element::i32, Shape{}, {0}));
     auto input = context.get_input(0);
     auto idx = context.get_input(1);
-    auto idx_unsqueezed = context.mark_node(std::make_shared<v0::Unsqueeze>(idx, zero));
+    auto rank = std::get<1>(get_shape_rank(context, input));
+    rank = context.mark_node(std::make_shared<v1::ConvertLike>(rank, idx));
+    idx = normalize_axis(context, idx, rank);
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/set_item.cpp:27]:
```suggestion
    auto shape = context.mark_node(std::make_shared<v3::ShapeOf>(input, element::i32));
    auto shape_0 = context.mark_node(std::make_shared<v8::Gather>(shape, zero, zero));
    shape_0 = context.mark_node(std::make_shared<v1::ConvertLike>(shape_0, idx));
    shape_0 = context.mark_node(std::make_shared<v0::Unsqueeze>(shape_0, zero));
    // ScatterUpdate-3 doesn't support negative indices
    idx = normalize_axis(context, idx, shape_0);
```

**Pull request closed without merging**